rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -101,6 +106,34 @@ void SqliteStorage::open(
     relative_path_ << "' for " << to_string(io_flag) << ".");
 }
 
+void SqliteStorage::activate_transaction()
+{
+  int rc = -1;
+
+  if (!active_transaction_) {
+    rc = sqlite3_exec(database_->get_db_handle(), "BEGIN TRANSACTION;", NULL, 0, NULL);
+    active_transaction_.store(true, std::memory_order_relaxed);
+    if (rc != SQLITE_OK)
+      throw SqliteException("Failed to initialize transaction");
+  }
+}
+
+void SqliteStorage::commit_transaction()
+{
+  int rc = -1;
+
+  if (!active_transaction_) {
+    rc = sqlite3_exec(database_->get_db_handle(), "END TRANSACTION;", NULL, 0, NULL);
```

Comment:
I believe this shouldn't be `END TRANSACTION` but rather `COMMIT`:
https://stackoverflow.com/questions/14806435/difference-between-end-transaction-and-commit-transaction